### DISCUSSION TITLE
Logical nullish assignment with Destructuring assignment

### DISCUSSION THREAD

**Discussion by chanelnumberseven** [babel/babel#13517] [Ideas]
Can i write code as follows
```
function demo({data??=[]}){
  return data;
}

demo({data:null}) // expect []
```

It is more beautiful than

```
function demo({data}){
  const data1=data??[];
  // use data1 do something
  return data1; 
}

demo({data:null}) // expect [] 
```

**Comment by nicolo-ribaudo** [comment]
No, that it not valid JS code and it not supported by any JS proposal.

I tried looking for past discussion about it and I only found https://esdiscuss.org/topic/null-coalescing-default-values-in-destructuring, but it looks like no one actually proposed that syntax to tc39.
